packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -336,15 +337,25 @@ const App = (props) => {
                                 <link
                                     rel="alternate"
                                     hrefLang={locale.id.toLowerCase()}
-                                    href={`${appOrigin}${buildUrl(location.pathname)}`}
+                                    href={`${appOrigin}${getPathWithLocale(locale.id, buildUrl, {
+                                        location: {
+                                            ...location,
+                                            search: ''
+                                        }
+                                    })}`}
                                     key={locale.id}
                                 />
                             ))}
                             {/* A general locale as fallback. For example: "en" if default locale is "en-GB" */}
                             <link
                                 rel="alternate"
                                 hrefLang={site.l10n.defaultLocale.slice(0, 2)}
-                                href={`${appOrigin}${buildUrl(location.pathname)}`}
+                                href={`${appOrigin}${getPathWithLocale(locale.id, buildUrl, {
+                                    location: {
+                                        ...location,
+                                        search: ''
+                                    }
+                                })}`}
                             />
                             {/* A wider fallback for user locales that the app does not support */}
                             <link rel="alternate" hrefLang="x-default" href={`${appOrigin}/`} />
```

Comment:
I believe there is a bug here as well... we are saying the default lang alt link is the home page all the time. No required for this ticket.. but I think we should capture this as a bug somewhere.

Reply:
@bendvc I looked into this link tag. I think what you meant is to have this link reflect current site location instead of keeping homepage all the time? I believe it should be the case since the purpose of x-default alt is to provide a link where no locale is matched. In our case, if someone accessed a locale link that is not available, the homepage is where it should load.